packages/datadog-plugin-rhea/src/index.js

```diff
@@ -118,30 +118,27 @@ function wrapDeliveryUpdate (update) {
 }
 
 function patchCircularBuffer (proto, instrumenter) {
-  Object.defineProperty(proto, 'session', {
+  Object.defineProperty(proto, 'outgoing', {
     configurable: true,
     get () {},
-    set (session) {
-      Object.defineProperty(this, 'session', {
+    set (outgoing) {
+      Object.defineProperty(this, 'outgoing', {
         configurable: true,
         writable: true,
         enumerable: true,
-        value: session
+        value: outgoing
       })
-      if (session) {
+      if (outgoing) {
         let CircularBuffer
-        if (session.outgoing && session.outgoing.deliveries) {
-          CircularBuffer = session.outgoing.deliveries.constructor
-        }
-        if (!CircularBuffer && session.incoming && session.incoming.deliveries) {
-          CircularBuffer = session.incoming.deliveries.constructor
+        if (outgoing.deliveries) {
+          CircularBuffer = outgoing.deliveries.constructor
```

Comment:
Patching the instance would make the code simpler. Since sessions are persistent, this won't affect the performance negatively.